services/wordpress/wordpress-author.service.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+const BaseWordpress = require('./wordpress-base')
+
+const extensionData = {
+  plugin: {
+    capt: 'Plugin',
+    exampleSlug: 'akismet',
+  },
+  theme: {
+    capt: 'Theme',
+    exampleSlug: 'twentytwenty',
+  },
+}
+
+function AuthorForType(extensionType) {
+  const { capt, exampleSlug } = extensionData[extensionType]
+
+  return class WordpressAuthor extends BaseWordpress {
+    static name = `Wordpress${capt}Author`
+
+    static category = 'social'
+
+    static route = {
+      base: `wordpress/${extensionType}/author`,
+      pattern: ':slug',
+    }
+
+    static examples = [
+      {
+        title: `WordPress ${capt} Author`,
+        namedParams: { slug: exampleSlug },
+        staticPreview: {
+          label: 'author',
+          message: 'wordpress',
+          style: 'social',
+        },
+      },
+    ]
+
+    static defaultBadgeData = {
+      label: 'author',
+      namedLogo: 'wordpress',
+    }
+
+    static render({ author, link }) {
+      return {
+        label: 'author',
+        message: author,
+        namedLogo: 'wordpress',
+        link,
+      }
+    }
+
+    transform(rawAuthor) {
+      return rawAuthor.substring(
+        rawAuthor.lastIndexOf('">') + 2,
+        rawAuthor.lastIndexOf('<')
+      )
+    }
```

Comment:
@chris48s - so this is a great example of the hypothetical scenario I referred to in https://github.com/badges/shields/pull/5722#discussion_r507268044 (the schema is just a `Joi.string().required()`)

I do feel strongly that _somewhere_ we need to have some kind of validation since we need to slice and dice the string value, whether we should do this as part of `transform` and throwing a more contextual error message or as part of the schema is where there's still some questions IMO.

Reply:
This is part of the service I do need to revisit actually as I realised it may not work all the time.

The string can contain a HTML link, which is what the transform is stripping out, leaving us with just the name of the author. However, if no link is set, I believe it should just return the name, no HTML. So I will need to rework this to detect  and deal with the HMTL and without correctly.

Reply:
Agreed. It makes sense to check the string contains `<` [some sensible number of characters] and a `>` before we try and parse it like it this, otherwise we're going to get unexpected outputs. Using a `.regex()` in the schema is probably the sensible place to do that.
Are we sure the string can contain only one HTML element?